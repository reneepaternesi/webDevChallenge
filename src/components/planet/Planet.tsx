// React
import React, { useEffect, useState } from 'react'
import { useLocation, useHistory } from 'react-router-dom'
// Styles
import GlobalStyle from '../../globalStyles'
import { PlanetWrapper, GridInfo, DataLine, DataName, TableData, BackButton } from './styles'

interface Props {}

const Planet: React.FC<Props> = () => {
  const [element, setElement] = useState<any | null>(null)
  const location = useLocation()
  const  history = useHistory();

  useEffect(() => {
    setElement(location.state)
  }, [location])

  const handleGoBack = () => {
    history.goBack()
  }

  return (
    <PlanetWrapper>
      <GlobalStyle />
      <h1>{element?.planet.name}</h1>
      <GridInfo>
        <DataLine>
          <DataName>Rotation Period:</DataName> {element?.planet.rotation_period}
        </DataLine>
        <DataLine>
          <DataName>Orbital Period:</DataName> {element?.planet.orbital_period}
        </DataLine>
        <DataLine>
          <DataName>Climate:</DataName> {element?.planet.climate}
        </DataLine>
        <DataLine>
          <DataName>Gravity:</DataName> {element?.planet.gravity}
        </DataLine>
        <DataLine>
          <DataName>Terrain:</DataName> {element?.planet.terrain}
        </DataLine>
        <DataLine>
          <DataName>Surface Water:</DataName> {element?.planet.surface_water}
        </DataLine>
        <DataLine>
          <DataName>Population:</DataName> {element?.planet.population}
        </DataLine>
        <DataLine />
        <DataLine>
          <DataName>Residents:</DataName>
          <TableData>
            <tbody>
              {element?.planet.residents.map((item: any, index: number) => (
                <tr key={index}>
                  <td>{item}</td>
                </tr>
              ))}
            </tbody>
          </TableData>
        </DataLine>
        <DataLine>
          <DataName>Films:</DataName>
          <TableData>
            <tbody>
              {element?.planet.films.map((item: any, index: number) => (
                <tr key={index}>
                  <td>{item}</td>
                </tr>
              ))}
            </tbody>
          </TableData>
        </DataLine>
      </GridInfo>
      <BackButton onClick={handleGoBack}> Go Back </BackButton>
    </PlanetWrapper>
  )
}

export default Planet
