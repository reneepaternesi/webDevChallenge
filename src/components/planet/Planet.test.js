import React from 'react'
import { shallow } from 'enzyme'
import Planet from './Planet'

const mockUseLocationValue = {
  pathname: '/planet',
  search: '',
  hash: '',
  state: null,
}

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useLocation: jest.fn().mockImplementation(() => {
    return mockUseLocationValue
  }),
}))

let wrapper

beforeEach(() => {
  wrapper = shallow(<Planet />)
})

describe('Planet Test', () => {
  it('should render Planet correctly', () => {
    mockUseLocationValue.state = {
      planet: {
        climate: 'temperate',
        created: '2014-12-10T11:50:29.349000Z',
        diameter: '4900',
        edited: '2014-12-20T20:58:18.429000Z',
        films: ['http://swapi.dev/api/films/3/'],
        gravity: '0.85 standard',
        name: 'Endor',
        orbital_period: '402',
        population: '30000000',
        residents: ['http://swapi.dev/api/people/30/'],
        rotation_period: '18',
        surface_water: '8',
        terrain: 'forests, mountains, lakes',
        url: 'http://swapi.dev/api/planets/7/',
      },
    }
    expect(wrapper).toMatchSnapshot()
  })
})
