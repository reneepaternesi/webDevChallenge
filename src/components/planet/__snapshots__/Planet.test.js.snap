// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet Test should render Planet correctly 1`] = `
<styled.div>
  <Memo(l) />
  <h1 />
  <styled.div>
    <styled.div>
      <styled.span>
        Rotation Period:
      </styled.span>
       
    </styled.div>
    <styled.div>
      <styled.span>
        Orbital Period:
      </styled.span>
       
    </styled.div>
    <styled.div>
      <styled.span>
        Climate:
      </styled.span>
       
    </styled.div>
    <styled.div>
      <styled.span>
        Gravity:
      </styled.span>
       
    </styled.div>
    <styled.div>
      <styled.span>
        Terrain:
      </styled.span>
       
    </styled.div>
    <styled.div>
      <styled.span>
        Surface Water:
      </styled.span>
       
    </styled.div>
    <styled.div>
      <styled.span>
        Population:
      </styled.span>
       
    </styled.div>
    <styled.div />
    <styled.div>
      <styled.span>
        Residents:
      </styled.span>
      <styled.table>
        <tbody />
      </styled.table>
    </styled.div>
    <styled.div>
      <styled.span>
        Films:
      </styled.span>
      <styled.table>
        <tbody />
      </styled.table>
    </styled.div>
  </styled.div>
  <styled.button
    onClick={[Function]}
  >
     Go Back 
  </styled.button>
</styled.div>
`;
